### PR TITLE
fix(Radio,Checkbox): fixes label onclick

### PR DESCRIPTION
**Pull request by stutirao**
### What does this implement/fix? Explain your changes.

...

### Does this close any currently open issues?

...

### Any other comments?

...

### Dependent PRs/Commits

...

### Describe breaking changes, if any.

...

### Checklist

Check all those that are applicable and complete.

- [ ] Merged with latest `master` branch

### CODE REVIEW

File: core/utils/UidGenerator.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export const UidGenerator = (name?: string, label?: string) => {
```
Comment (aditya-kumawat) [inline core/utils/UidGenerator.tsx]: No need to pass arguments.

File: core/utils/UidGenerator.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  let dt = new Date().getTime();
+  const uuid = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, c => {
+    /* tslint:disable:no-bitwise */
+    const r = (dt + Math.random() * 16) % 16 | 0;
+    dt = Math.floor(dt / 16);
+    const s = (c === 'x' ? r : (r & 0x3 | 0x8)).toString(16);
+    return (`${name}${'-'}${label}${'-'}${s}`);
```
Comment (aditya-kumawat) [inline core/utils/UidGenerator.tsx]: Return randomly generated string only.

File: core/components/atoms/checkbox/Checkbox.tsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import classNames from 'classnames';
 import Text from '@/components/atoms/text';
 import Icon from '@/components/atoms/icon';
 import { BaseProps, extractBaseProps } from '@/utils/types';
-
+import UidGenerator from '../../../utils/UidGenerator';
```
Comment (aditya-kumawat) [inline core/components/atoms/checkbox/Checkbox.tsx]: '../../../utils/UidGenerator' => '@/utils/UidGenerator'

File: core/components/atoms/checkbox/Checkbox.tsx [hunk 4]
```diff
@@ -128,7 +128,8 @@ export const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>((props
     }
     if (onChange) onChange(e);
   };
-
+  let id = UidGenerator();
+  id = `${name}-${label}-${id}`;
```
Comment (aditya-kumawat) [inline core/components/atoms/checkbox/Checkbox.tsx]: Please do it in single line.
const id = \`${name}-${label}-${UidGenerator()}\`;

File: core/components/atoms/radio/Radio.tsx [hunk 5]
```diff
@@ -79,6 +80,8 @@ export const Radio = React.forwardRef<HTMLInputElement, RadioProps>((props, forw
     if (onChange) onChange(event);
   };
 
+  let id = UidGenerator();
+  id = `${name}-${label}-${id}`;
```
Comment (aditya-kumawat) [inline core/components/atoms/radio/Radio.tsx]: Please do it in single line.

File: core/components/atoms/checkbox/Checkbox.tsx [hunk 6]
```diff
@@ -3,7 +3,7 @@ import classNames from 'classnames';
 import Text from '@/components/atoms/text';
 import Icon from '@/components/atoms/icon';
 import { BaseProps, extractBaseProps } from '@/utils/types';
-
+import UidGenerator from '@/utils/UidGenerator';
```
Comment (aditya-kumawat) [inline core/components/atoms/checkbox/Checkbox.tsx]: UidGenerator => uidGenerator

**Comment by aditya-kumawat** [comment]
@stutirao Also update the name of the file to uidGenerator. Not just the import name.

**Review by aditya-kumawat — DISMISSED** [review]

**Review by aditya-kumawat — APPROVED** [review]

**Review by satyamyadav — APPROVED** [review]

**Pull request merged**